### DISCUSSION TITLE
Why the results are different between R and GSEApy

### DISCUSSION THREAD

**Discussion by sinorockie** [zqfang/GSEApy#284] [Q&A]
In R language
`
library(GSVA)
library(GSVAdata)
params <- ssgseaParam(exprData = exp4, geneSets = genesets)
gsva_results <- gsva(params)
`
In GSEpy
`
ssgsea_kwargs = {
    "min_size": 1,
    "max_size": exp4.shape[0],
    "weight": 0.25,
    "verbose": True,
    "seed": 1
}
ss = gseapy.ssgsea(data=data, gene_sets=gene_sets, **ssgsea_kwargs)
gsva_results = ss.res2d.pivot(index='Term', columns='Name', values='NES')
`

I'm trying to reproduce the result in GSEApy, however, the result is different from the result from R, I have check the input data and genesets which are same.

Is there any one can tell me the reason?

**Comment by zqfang** [comment]
I think there are some filtering steps before real calculation.

see [here]( https://github.com/zqfang/GSEApy/blob/5e65a96bdec03073e46321c736d0cad1c0d66e56/src/gsva.rs#L342) for testing codes. if you use the same input files for both gseapy and GSVA. They have the same output
